pkg/3scale/amp/operator/upgrade.go

```diff
@@ -191,21 +192,25 @@ func (u *UpgradeApiManager) upgradeImages() (reconcile.Result, error) {
 		return res, err
 	}
 
-	if !u.apiManager.IsExternalDatabaseEnabled() {
-		res, err = u.upgradeBackendRedisImageStream()
-		if res.Requeue || err != nil {
-			return res, err
-		}
+	upgradeImages := sequentially(
+		when(
```

Comment:
This is actually not needed because in the upgrade procedure, customers could not specify external dependencies. It was all or nothing (expect for the zync database). But it should also work